src/components/ContactsPage.js

```diff
@@ -43,6 +43,13 @@ class ContactsPage extends React.Component {
     this.setState({ contacts });
   }
 
+  deleteContact(id) {
+    const { contacts } = this.state;
+    const index = contacts.findIndex(item => item.id === id);
+    contacts.splice(index, 1);
```

Comment:
The splice() function is mutating the array (which is never a good approach). And you are also parsing the array 2 times: once for getting the index and once for finding and removing the element at that index. Here is a better, non-mutating solution:

```
const updatedContacts = contacts.filter(contact => contact.id !== id);
this.setState({ contacts: updatedContacts });
```